pkg/util/parallel/parallel.go

```diff
@@ -0,0 +1,27 @@
+package parallel
+
+import (
+	"sync"
+)
+
+// Run executes the provided functions in parallel and collects any errors they return.
+func Run(fns ...func() error) []error {
+	wg := sync.WaitGroup{}
+	errCh := make(chan error, len(fns))
+	wg.Add(len(fns))
+	for i := range fns {
+		go func(i int) {
+			if err := fns[i](); err != nil {
+				errCh <- err
+			}
+			wg.Done()
+		}(i)
+	}
+	wg.Wait()
+	close(errCh)
+	var errs []error
+	for err := range errCh {
```

Comment:
Range exits when channel is closed and empty.  Although now I want to do a test to prove that...

> On Jun 9, 2015, at 10:23 PM, Derek Carr <notifications@github.com> wrote:
> 
> In pkg/util/parallel/parallel.go:
> 
> > +func Run(fns ...func() error) []error {
> > +	wg := sync.WaitGroup{}
> > +	errCh := make(chan error, len(fns))
> > +	wg.Add(len(fns))
> > +	for i := range fns {
> > +		go func(i int) {
> > +			if err := fns[i](); err != nil {
> > +				errCh <- err
> > +			}
> > +			wg.Done()
> > +		}(i)
> > +	}
> > +	wg.Wait()
> > +	close(errCh)
> > +	var errs []error
> > +	for err := range errCh {
> Does it check the chan is closed? Or can you range and block on an open chan?
> 
> —
> Reply to this email directly or view it on GitHub.
>